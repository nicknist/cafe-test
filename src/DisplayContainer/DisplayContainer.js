import React from 'react';
import Reservation from '../Reservation/Reservation';

const DisplayContainer = ({ reservations, deleteReservation }) => {
  if (reservations.length === 0) {
    return <h1>Loading...</h1>
  }
  let reservationList = reservations.map(res => {
    return <Reservation deleteReservation={deleteReservation} key={res.name} className='single-res' reservation={res}/>
  })

  return (
    <div className='resy-container'>
      {reservationList}
    </div>
  )
}

export default DisplayContainer;
