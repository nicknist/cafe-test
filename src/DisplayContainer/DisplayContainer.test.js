import React from 'react';
import ReactDOM from 'react-dom';
import DisplayContainer from './DisplayContainer';
import { shallow } from 'enzyme';

describe('DisplayContainer', () => {
  let wrapper;
  let mockReservations;
  let res1;
  let res2;

  beforeEach(() => {
    res1 = {
      name: 'Uncle Robbie',
      date: '12-11',
      time: '12:20',
      number: 200
    }
    res2 = {
      name: 'Dad Eric',
      date: '12-25',
      time: '4:00',
      number: 1
    }
    mockReservations = [res1, res2]
    wrapper = shallow(<DisplayContainer reservations={mockReservations}/>);
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  })
})
