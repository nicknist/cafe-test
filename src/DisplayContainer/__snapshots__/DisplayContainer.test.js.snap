// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DisplayContainer should match the snapshot 1`] = `
<div
  className="resy-container"
>
  <Reservation
    className="single-res"
    key="Uncle Robbie"
    reservation={
      Object {
        "date": "12-11",
        "name": "Uncle Robbie",
        "number": 200,
        "time": "12:20",
      }
    }
  />
  <Reservation
    className="single-res"
    key="Dad Eric"
    reservation={
      Object {
        "date": "12-25",
        "name": "Dad Eric",
        "number": 1,
        "time": "4:00",
      }
    }
  />
</div>
`;
